### PR TITLE
[bitnami/nginx] possibility to add lifecycle hooks

### PR DESCRIPTION
**Pull request by Adrian43211234**
**Description of the change**

Add the possibility to add lifecycle hooks: https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/
e.g. https://news.ycombinator.com/item?id=30771149  it prevents ingress routes to send requests to the terminating nginx pods which would else result in 502 errors.

**Benefits**
Following snippet provides a 100% uptime during a deployment restart of nginx with 2 replicas. With this configuration, the termination of the nginx pod gets delayed until the ingress controller removed that container as endpoint.
```
lifecycle:
  preStop:
    exec:
      command: ["/bin/sleep", "20"]
```
tested with 

`siege -v -H "Authorization: Bearer XXXXXXX" <URL> -c 2`

**Possible drawbacks**

Implementation of `{{- toYaml .Values.lifecycle | nindent 12 }}` fits best for different use-cases from my point of view but also could end in syntax errors if configured incorrectly.

**Applicable issues**

  - fixes #

**Additional information**

**Checklist**

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by Adrian43211234** [comment]
?

**Review by FraPazGal — COMMENTED** [review]
Hello @Adrian43211234,

We're sorry about the delay, it seems this PR got lost due to an error in our support workflow. We'll check it in case we missed anything else. Thanks for the patience!

Regarding the change itself, could you please name the parameter as `lifecycleHooks`. This is to maintain the same [template naming](https://github.com/bitnami/charts/blob/4a94de4eafacc50019cc702ccf1c0bb688ec0937/template/CHART_NAME/templates/deployment.yaml#L150) throughout our catalog.

**Comment by Adrian43211234** [comment]
Hmm... I believe I signed the commits, what is the problem now?

File: bitnami/nginx/templates/deployment.yaml [hunk 1]
```diff
@@ -173,6 +173,9 @@ spec:
             - name: https
               containerPort: {{ .Values.containerPorts.https }}
             {{- end }}
+          {{- if .Values.lifecycleHooks }}
+          lifecycle: {{- toYaml .Values.lifecycleHooks | nindent 12 }}
```
Comment (FraPazGal) [inline bitnami/nginx/templates/deployment.yaml:177]: I saw you mentioned using `toYaml` here and its possible drawbacks. Could you please check if using our default `common.tplvalues.render` function will work better here?

```
{{- define "common.tplvalues.render" -}}
    {{- if typeIs "string" .value }}
        {{- tpl .value .context }}
    {{- else }}
        {{- tpl (.value | toYaml) .context }}
    {{- end }}
{{- end -}}
```

Referring again to our scaffolding templates, we normally use:

```suggestion
          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.lifecycleHooks "context" $) | nindent 12 }}
```

**Review by FraPazGal — COMMENTED** [review]
Looking at the [action's summary](https://github.com/bitnami/charts/pull/9609/checks?check_run_id=6004063623), it seems it was expecting a different email to be used to sign-off one of the commits. Could it be related to some changes in your GH emails? Nevertheless, seeing as every commit is correctly I'll override the blocker when merging the PR.

I added one last comment to the PR before approving it.

File: bitnami/nginx/templates/deployment.yaml (hunk 1 above)
Comment (Adrian43211234) [reply bitnami/nginx/templates/deployment.yaml]: yes it does work with the default as well

grafik

**Review by FraPazGal — APPROVED** [review]
Thanks for the PR, LGTM!

**Pull request merged**